contracts/factory/src/storage.rs

```diff
@@ -30,6 +30,27 @@ pub struct Config {
     pub multihop_address: Address,
 }
 
+#[contracttype]
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct LpAndStakeInfo {
```

Comment:
How about:
```suggestion
pub struct UserPortfolio {
```
the rest can stay as it is.